server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -519,13 +464,12 @@ public MappingMetaData mappingOrDefault(String mappingType) {
         return mappings.get(MapperService.DEFAULT_MAPPING);
     }
 
-    public ImmutableOpenMap<String, Custom> getCustoms() {
-        return this.customs;
+    public ImmutableOpenMap<String, DiffableStringMap> getCustomData() {
+        return this.customData;
     }
 
-    @SuppressWarnings("unchecked")
-    public <T extends Custom> T custom(String type) {
-        return (T) customs.get(type);
+    public Map<String, String> getCustomData(final String key) {
```

Comment:
Are we returning here a mutable map? I don't think we should do that here. We should at least wrap it in `Collections#unmodifiableMap()`

Reply:
Yep good idea, I'll do that